crates/modelardb_server/tests/integration_test.rs

```diff
@@ -586,22 +613,38 @@ fn test_can_list_actions() {
 fn test_can_collect_metrics() {
     let mut test_context = TestContext::new();
 
-    let metrics = test_context.runtime.block_on(async {
-        test_context
-            .client
-            .do_action(Request::new(Action {
-                r#type: "CollectMetrics".to_owned(),
-                body: Bytes::new(),
-            }))
-            .await
-            .unwrap()
-            .into_inner()
-            .map(|metric| metric.unwrap().body)
-            .collect::<Vec<Bytes>>()
-            .await
-    });
+    // Ingest data points and flush the edge to populate all currently collected metrics.
+    let time_series = TestContext::generate_time_series_with_tag(false, None, Some("tag"));
+    let flight_data = TestContext::create_flight_data_from_time_series(
+        TABLE_NAME.to_owned(),
+        &[time_series.clone()],
+    );
+
+    test_context.create_table(TABLE_NAME, TableType::ModelTable);
 
-    assert!(!metrics.is_empty());
+    test_context
+        .send_time_series_to_server(flight_data)
+        .unwrap();
+
+    test_context.flush_data_to_disk();
+
+    // Collect the metrics.
+    let metrics = test_context.retrieve_action_record_batch("CollectMetrics");
+
+    // Check that all metrics are present in the response.
+    let metrics_array = modelardb_common::array!(metrics, 0, StringArray);
+
+    assert_eq!(metrics_array.value(0), "used_uncompressed_memory");
+    assert_eq!(metrics_array.value(1), "used_compressed_memory");
+    assert_eq!(metrics_array.value(2), "ingested_data_points");
+    assert_eq!(metrics_array.value(3), "used_disk_space");
+
+    // Check that the metrics are populated when ingesting and flushing.
+    let timestamps_array = modelardb_common::array!(metrics, 1, ListArray);
+    assert_eq!(timestamps_array.value(0).len(), 2);
```

Comment:
Maybe match the specific lists when possible and document why only their length can be checked for the rest.

Reply:
Good idea, fixed.